numpy/core/fromnumeric.py

```diff
@@ -70,11 +70,29 @@ def take(a, indices, axis=None, out=None, mode='raise'):
     using arrays); however, it can be easier to use if you need elements
     along a given axis.
 
+    When axis is not None this is equivalent to the following use of `ndindex`,
+    which sets each of ``ii``, ``jj``, and ``kk`` to a tuple of indices::
+
+        Ni, Nk = a.shape[:axis], a.shape[axis+1:]
+        Nj = indices.shape
+        for ii in ndindex(Ni):
+            for jj in ndindex(Nj):
+                for kk in ndindex(Nk):
+                    out[ii + jj + kk] = a[ii + (indices[jj],) + kk]
+
+    Equivalently, using `s_` to build simple slice objects, this can be
+    expressed in terms of applying fancy indexing to each 1d slice::
+
+        Ni, Nk = a.shape[:axis], a.shape[axis+1:]
+        for ii in ndindex(Ni):
+            for kk in ndindex(Nj):
+                out[ii + s_[...,] + kk] = a[ii + s_[:,] + kk][indices]
```

Comment:
As in the above comment, could be:
```python
out[(*ii, ..., *kk)] = a[(*ii, s_[:], *kk)]
```

Reply:
I think there is a little less mental overhead for tuple-addition than for tuple-unpacking, especially for people new to python.